sec_certs_page/dashboard/callbacks/charts.py

```diff
@@ -0,0 +1,232 @@
+from typing import TYPE_CHECKING
+
+import dash_bootstrap_components as dbc
+from dash import ALL, MATCH, ctx, html, no_update
+from dash.dependencies import Input, Output, State
+from dash.development.base_component import Component
+
+from ..chart.chart import Chart
+from ..chart.factory import ChartFactory
+from ..types.common import CollectionName
+from .utils import create_chart_wrapper
+
+if TYPE_CHECKING:
+    from ...common.dash.base import Dash
+    from ..chart.registry import ChartRegistry
+    from ..data import DataService
+
+
+def register_chart_callbacks(
+    dash_app: "Dash",
+    prefix: str,
+    chart_registry: "ChartRegistry",
+    data_service: "DataService",
+) -> None:
+    _register_chart_management(dash_app, prefix, chart_registry)
+    _register_predefined_chart_options(dash_app, prefix, chart_registry)
+    _register_chart_rendering(dash_app, prefix, chart_registry, data_service)
+    _register_update_all(dash_app, prefix)
+
+
+def register_pattern_matching_callbacks(
+    dash_app: "Dash",
+    chart_registries: dict[CollectionName, "ChartRegistry"],
+) -> None:
+    """Must be registered once globally since pattern-matching callbacks match across all collections."""
+
+    @dash_app.callback(
+        output=dict(content=Output({"type": "chart-content", "index": MATCH}, "children")),
+        inputs=dict(n_clicks=Input({"type": "chart-refresh", "index": MATCH}, "n_clicks")),
+        state=dict(wrapper_id=State({"type": "chart-wrapper", "index": MATCH}, "id")),
+        prevent_initial_call=True,
+    )
+    def refresh_single_chart(n_clicks, wrapper_id):
```

Comment:
Maybe a docstring here?
```suggestion
    def refresh_single_chart(n_clicks, wrapper_id):
        """When user clicks the chart-refresh button we re-render it."""
```